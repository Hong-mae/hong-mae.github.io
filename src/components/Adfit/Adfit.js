// @flow strict
import React, { useEffect } from 'react';
import classNames from 'classnames';
import styles from './Adfit.module.scss';

type Props = {
    name: string,
    pc: string,
    mobile: string,
    width?: string,
    height?: string,
};

const Adfit = ({ name, pc, mobile, width = '728', height = '90' }: Props) => {
    useEffect(() => {
        const ins = document.createElement('ins');
        const script = document.createElement('script');

        ins.className = 'kakao_ad_area';
        ins.style.display = 'none';
        ins.style.width = '100%';

        if (window.innerWidth < 756) {
            ins.setAttribute('data-ad-unit', mobile);
        } else {
            ins.setAttribute('data-ad-unit', pc);
        }
        ins.setAttribute('data-ad-width', width);
        ins.setAttribute('data-ad-height', height);

        script.src = '//t1.daumcdn.net/kas/static/ba.min.js';
        script.async = true;
        script.type = 'text/javascript';

        const adfitArea = document.querySelector(`div.${name}.ad_area`);
        if (adfitArea) {
            adfitArea.append(ins);
            adfitArea.append(script);
        }
    }, []);

    return (
        <div className={styles['adfit']}>
            <div className={classNames(name, 'ad_area')}></div>
        </div>
    );
};

export default Adfit;
