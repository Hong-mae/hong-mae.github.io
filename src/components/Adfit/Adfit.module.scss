@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.adfit {
  text-align: center;
}
