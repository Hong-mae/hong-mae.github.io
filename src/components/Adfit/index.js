// @flow strict
export { default } from './Adfit';
