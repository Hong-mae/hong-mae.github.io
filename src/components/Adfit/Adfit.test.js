// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Adfit from './Adfit';

describe('Adfit', () => {
    const props = {
        name: 'adfit_test_js',
        pc: 'pctest',
        mobile: 'mbtest',
    };

    it('render correctly', () => {
        const tree = renderer.create(<Adfit {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
