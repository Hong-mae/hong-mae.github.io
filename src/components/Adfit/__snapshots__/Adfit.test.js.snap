// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adfit render correctly 1`] = `
<div
  className="adfit"
>
  <div
    className="adfit_test_js ad_area"
  />
</div>
`;
